lstchain/scripts/onsite/onsite_create_cat_B_calibration_file.py

```diff
@@ -0,0 +1,223 @@
+    '--sys_date',
+    help=(
+        "Date of systematic correction file (format YYYYMMDD). \n"
+        "Default: automatically search the best date \n"
+    ),
+)
+optional.add_argument('--no_sys_correction',
+                      help="Systematic corrections are not applied. \n",
+                      action='store_true',
+                      default=False)
+optional.add_argument('--output_base_name', help="Base of output file name (change only for debugging)",
+                      default="calibration")
+
+optional.add_argument('--n_subruns', help="Number of subruns to be processed",
+                      type=int, default=MAX_SUBRUNS)
+
+optional.add_argument('-f', '--filters', help="Calibox filters")
+optional.add_argument('--tel_id', help="telescope id. Default = 1", type=int, default=1)
+
+optional.add_argument('--config', help="Config file", default=DEFAULT_CONFIG_CAT_B_CALIB, type=Path)
+optional.add_argument('--mongodb', help="Mongo data-base connection", default="mongodb://10.200.10.161:27018/")
+
+optional.add_argument('-y', '--yes', action="store_true", help='Do not ask interactively for permissions, assume true')
+optional.add_argument('--no_pro_symlink', action="store_true",
+                      help='Do not update the pro dir symbolic link, assume true')
+
+
+def main():
+    args, remaining_args = parser.parse_known_args()
+    run = args.run_number
+    n_subruns = args.n_subruns
+    prod_id = args.prod_version
+    stat_events = args.statistics
+    
+    sys_date = args.sys_date
+    no_sys_correction = args.no_sys_correction
+    tel_id = args.tel_id
+    config_file = args.config
+    yes = args.yes
+    pro_symlink = not args.no_pro_symlink
+    r0_dir = args.r0_dir or args.base_dir / 'R0'
+    dl1_dir = args.dl1_dir or args.base_dir / 'DL1'
+
+
+    # looks for the filter values in the database if not given
+    if args.filters is None:
+        filters = find_filter_wheels(run, args.mongodb)
+    else:
+        filters = args.filters
+
+    if filters is None:
+        sys.exit(f"Missing filter value for run {run}. \n")
+
+    print(f"\n--> Start calculating Cat-B calibration from run {run}, filters {filters}")
+
+    # verify config file
+    if not config_file.exists():
+        raise IOError(f"Config file {config_file} does not exists. \n")
+
+    print(f"\n--> Config file {config_file}")
+
+   # verify input file
+    input_files = find_interleaved_subruns(run, r0_dir, dl1_dir)
+    input_path = input_files[0].parent
+    print(f"\n--> Found {len(input_files)} interleaved subruns in {input_path}")
+    if n_subruns < MAX_SUBRUNS:
+        print(f"--> Process {n_subruns} subruns")
+
+    # verify output dir
+    date = input_files[0].parents[1].name
+    calib_dir = args.base_dir / CAT_B_PIXEL_DIR
+    output_dir = calib_dir / "calibration" / date / prod_id
+    if not output_dir.exists():
+        print(f"\n--> Create directory {output_dir}")
+        output_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
if `parents=True`, `exist_ok=True` are used, the previous conditional can be omitted.

Reply:
mmh, it seems, not really the same https://csatlas.com/python-create-directory/

Reply:
I do not see why it is different. If the dir exists and `exit_ok=True` it does nothing. If it does not exist, it creates it. But again, just a very minor detail.